File: string/reverse-vowels-of-a-string/reverse-vowels-of-a-string.ts
function reverseVowels(s: string): string {
    const ans = s.split("");
    const vowels = new Set(["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]);

    let i = 0;
    let j = s.length - 1;
    while (i < j) {
        while (i < j && !vowels.has(ans[i])) {
            i++;
        }
        while (i < j && !vowels.has(ans[j])) {
            j--;
        }
        let aux = s[i];
        ans[i++] = ans[j];
        ans[j--] = aux;
    }

    return ans.join('');
};

File: string/reverse-vowels-of-a-string/reverse-vowels-of-a-string.go
package reversevowelsofastring

func reverseVowels(s string) string {
	vowels := map[byte]bool{
		'a': true,
		'e': true,
		'i': true,
		'o': true,
		'u': true,
		'A': true,
		'E': true,
		'I': true,
		'O': true,
		'U': true,
	}
	ans := []byte(s)

	i := 0
	j := len(ans) - 1
	for i < j {
		for i < j && !vowels[ans[i]] {
			i++
		}
		for i < j && !vowels[ans[j]] {
			j--
		}

		aux := ans[i]
		ans[i] = ans[j]
		ans[j] = aux
		i++
		j--
	}

	return string(ans)
}
